src/MongoDB.EntityFrameworkCore/Storage/MongoDatabaseWrapper.cs

```diff
@@ -213,20 +203,7 @@ private static MongoUpdate ConvertModifiedEntryToMongoUpdate(string collectionNa
         var entityType = entry.EntityType;
         var entitySerializer = (IBsonDocumentSerializer)EntitySerializer.Create(entityType);
 
-        var fieldValues = new BsonDocument();
-        foreach (var property in entityType.GetProperties())
-        {
-            if (entry.IsModified(property))
-            {
-                var propertyValue = entry.GetCurrentValue(property);
-                var propertySerializationInfo = GetPropertySerializationInfo(entitySerializer, property);
-                var propertySerializer = propertySerializationInfo.Serializer;
-                var serializedPropertyValue = SerializeValue(propertySerializer, propertyValue);
-
-                var elementName = propertySerializationInfo.ElementName;
-                fieldValues[elementName] = serializedPropertyValue;
-            }
-        }
+        var fieldValues = ToBsonDocument(entry, entitySerializer, entityType.GetProperties().Where(p => entry.IsModified(p)));
```

Comment:
Did you try serializing without the PKs not additionally outside of _id? It would be cool to know if we can actually just "trick" the serializing into always putting them in the _id and not duplicate them.

Reply:
We are duplicating them temporarily, as LINQ provider does not aware about them being moved inside _id. Will rework this once Robert done his part in Driver.